src/net/metaplaylist/index.ts

```diff
@@ -0,0 +1,407 @@
+ * Adaptation and Representation from base content.
+ */
+function getParserBaseArguments<T>(
+  segment: ISegment,
+  metaplaylistArguments: ISegmentParserArguments<T>,
+  offset?: number
+): ISegmentParserArguments<T>;
+function getParserBaseArguments(
+  segment: ISegment,
+  metaplaylistArguments: ISegmentLoaderArguments,
+  offset?: number
+): ISegmentLoaderArguments;
+function getParserBaseArguments<T>(
+  segment: ISegment,
+  metaplaylistArguments: ISegmentParserArguments<T>|ISegmentLoaderArguments,
+  offset?: number
+): ISegmentParserArguments<T>|ISegmentLoaderArguments {
+  const originalSegment =
+    offset ?
+      objectAssign(
+        {},
+        segment,
+        { time: segment.isInit ? segment.time : segment.time - offset }
+      ) :
+      segment;
+
+  if (segment.privateInfos && segment.privateInfos.baseContentInfos) {
+    const contentInfos = segment.privateInfos.baseContentInfos;
+    return objectAssign({}, metaplaylistArguments, {
+      manifest: contentInfos.manifest,
+      period: contentInfos.period || metaplaylistArguments.period,
+      adaptation: contentInfos.adaptation || metaplaylistArguments.adaptation,
+      representation: contentInfos.representation || metaplaylistArguments.representation,
+      segment: originalSegment,
+    });
+  } else {
+    return objectAssign({}, metaplaylistArguments, { segment: originalSegment });
+  }
+}
+
+/**
+ * Get transport type from segment's privateInfos
+ * @param {Object} privateInfos
+ */
+function getTransportTypeFromSegmentPrivateInfos(
+  privateInfos: IPrivateInfos
+): ITransportTypes {
+  const transportType = privateInfos.transportType;
+
+  if (!transportType) {
+    throw new Error("Undefined transport for content for metaplaylist.");
+  }
+
+  return transportType;
+}
+
+export default function(options?: ITransportOptions): ITransportPipelines {
+
+    const transports = {
+      dash: DASHTransport(options),
+      smooth: SmoothTransport(options),
+    };
+
+    const manifestPipeline = {
+      loader(
+        { url } : IManifestLoaderArguments
+      ) : ILoaderObservable<Document|string> {
+        return request({
+          url,
+          responseType: "text",
+          ignoreProgressEvents: true,
+        });
+      },
+
+      parser(
+        { response, url } : IManifestParserArguments<Document|string>
+      ) : IManifestParserObservable {
+        if (typeof response.responseData !== "string") {
+          throw new Error("Parser input must be string.");
+        }
+        const { contents, attributes } = parseMetaPlaylistData(response.responseData);
+        const contents$ = contents.map((content) => {
+          const transport = transports[content.transport];
+          if (transport == null) {
+            throw new Error(
+              "Transport" + content.transport + "not supported in MetaPlaylist.");
+          }
+          const loader = transport.manifest.loader;
+
+          return loader({ url: content.url })
```

Comment:
Why don't we do that in the loader again? This seems safer + we have all the backoff thingies and all.